include/ensmallen_bits/pso/update_policies/lbest_update.hpp

```diff
@@ -0,0 +1,133 @@
+/**
+ * @file lbest_update.hpp
+ * @author Chintan Soni
+ * @author Suryoday Basak
+ *
+ * The local best version of PSO in which particles communicate with only two
+ * neighbours each, thus forming a ring topology amongst them. This approach
+ * allows PSO to converge at the global minimum, but takes significantly more
+ * iterations to do so.
+ *
+ * ensmallen is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef ENSMALLEN_PSO_UPDATE_POLICIES_LBEST_UPDATE_HPP
+#define ENSMALLEN_PSO_UPDATE_POLICIES_LBEST_UPDATE_HPP
+#include <assert.h>
+
+namespace ens {
+
+/**
+ * ADD SLIGHTLY MORE DETAILED DESCRIPTION HERE.
+ *
+ * For more information, see the following.
+ *
+ * REFERENCE TO A RELEVANT ARTICLE.
+ *
+ */
+class LBestUpdate
+{
+ public:
+  /**
+   * The Initialize method is called by PSO Optimizer method before the start of
+   * the iteration process. It calculates the value of the constriction
+   * coefficent, initializes the local best indices of each particle to itself,
+   * and sets the shape of the r1 and r2 vectors.
+   *
+   * @param exploitationFactor Influence of personal best achieved.
+   * @param explorationFactor Influence of neighbouring particles.
+   * @param numParticles The number of particles in the swarm.
+   * @param iterate The user input, used for shaping intermediate vectors.
+   */
+  void Initialize(const double& exploitationFactor,
+                  const double& explorationFactor,
+                  const size_t& numParticles,
+                  const arma::mat& iterate)
+  {
+    // Set number of particles.
+    n = numParticles;
```

Comment:
Don't see a reason to make an alias for ``numParticles``, ``exploitationFactor`` and ``explorationFactor``. In fact, I think the other parameter names are much more descriptive.

Reply:
I'll resolve this.

Reply:
The reason the aliases are used this way is that the variable names `c1`, `c2`, etc. have these usual meanings in literature.

Reply:
True, probably a good idea to go with the common naming scheme.